### PR TITLE
HPCC-28601 AddFileScope on AAD reports User Account Already Exists

### PR DESCRIPTION
**Pull request by RussWhitehead**
Because AAD adds the current user to a newly created FileScope OU, it is not an error that that user's ACE entries exist in the Access Control List (ACL). This PR adds some comments to the existing code, and changes the exception thrown when the user ACE exists into an information debug message.

Signed-off-by: Russ Whitehead <russwhitehead@yahoo.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/security/LdapSecurity/permissions.cpp [hunk 1]
```diff
@@ -1247,6 +1247,11 @@ bool PermissionProcessor::getPermissionsArray(CSecurityDescriptor *sd, IArrayOf<
 
 CSecurityDescriptor* PermissionProcessor::changePermission(CSecurityDescriptor* initialsd, CPermissionAction& action)
 {
+#ifdef _DEBUG
+	DBGLOG("changePermission- Action:%s, BaseDN:%s, ResType:%d, ResName:%s, AcctName:%s, AcctType:%d, Allows:%d, Denies:%d",
```
Comment (wangkx) [inline system/security/LdapSecurity/permissions.cpp:1251]: wrong indent

File: system/security/LdapSecurity/permissions.cpp [hunk 2]
```diff
@@ -1322,18 +1328,20 @@ CSecurityDescriptor* PermissionProcessor::changePermission(CSecurityDescriptor*
...
             PSID cursid = (PSID)&(pAce->SidStart);
 
-            if(EqualSid(cursid, act_psid))
+            if(EqualSid(cursid, act_psid))              //Check for ACE to be replaced
             {
+#ifdef _DEBUG
                 if(stricmp(action.m_action.str(), "add") == 0)
-                    throw MakeStringException(-1, "Permission for account %s already exists", action.m_account_name.str());
-                DeleteAce(pdacl, iAce);
+                    DBGLOG("Permission for account %s already exists", action.m_account_name.str());
```
Comment (wangkx) [inline system/security/LdapSecurity/permissions.cpp:1340]: When calling the 'add' action and the permission for account already exists, what will LDAP do (overwrite?, skip?, or ?) ?

**Review by wangkx — COMMENTED** [review]
@RussWhitehead 1 minor comment and 1 question.

File: system/security/LdapSecurity/permissions.cpp (hunk 2 above)
Comment (RussWhitehead) [reply system/security/LdapSecurity/permissions.cpp:1340]: For an "add" the previous permission will be removed, and the new (added) permission will be granted. 
This is different than an "update:," where the permissions would be merged

Comment (wangkx) [reply system/security/LdapSecurity/permissions.cpp:1340]: Got it

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine

**Comment by RussWhitehead** [comment]
@richardkchapman @ghalliday Please merge

**Pull request merged**